ApollosProject/apollos-prototype
# Adds custom theme scaffolding

## DESCRIPTION
This PR is an extension to #735 and while minimally functional on its own should be merged after it.

### What does this PR do, or why is it needed?
This establishes a `/theme` directory in the church/organization specific package (e.g. `apolloschurchapp`) and provides some structure to a few areas of app customization that will likely continue to grown and evolve. One of my favorite parts about what I am purposing in this PR is that we effectively are establishing and modeling the same folder structure as `/apollos-ui-kit/src/theme` but in app package. This should help provide guidance and understanding as to how this all works by having a strong example in `/apollos-ui-kit`.

##### Here is what happened
- the ui-kit `/icons` directory, where specific icon components (e.g. `BrandIcon`) live, is now in the `/apollos-ui-kit/src/theme` directory.
- custom icons follow the same paradigm above in `/apolloschurchapp/src/theme/icons`
- a custom theme definition now live in `/apolloschurchapp/src/theme` as well.
- an empty and well documented example `theme.js` is provided and already imported in `Providers` and passed into the `themeInput`. Providing custom theme attributes is a one step, plug-n-play, turn key, simplistic buzz word analogy process.
- custom icons are pre-configured the same as above. Just put them in the `icons` directory and export them from the `index` 💥

### What design trade-offs/decisions were made?
🤷‍♂

### How do I test this PR?
`yarn start` make sure nothing is broken and things look right. Add an icon component to the folder or add some custom theme definitions if you want but there are snapshot test that cover both of these situations already.

### What automated tests did you write?
None. Everything that was testable was already covered.

## TODO:

- [x] Partial work towards #558 

## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
